src/bun.js/test/jest.zig

```diff
@@ -1486,16 +1487,63 @@ pub const DescribeScope = struct {
         },
     );
 
+    pub fn onDone(
+        ctx: js.JSContextRef,
+        callframe: *JSC.CallFrame,
+    ) callconv(.C) JSValue {
+        const function = callframe.callee();
+        const args = callframe.arguments(1);
+        defer ctx.bunVM().autoGarbageCollect();
+
+        if (JSC.getFunctionData(function)) |data| {
+            var scope = bun.cast(*DescribeScope, data);
+            JSC.setFunctionData(function, null);
+            if (args.len > 0) {
+                const err = args.ptr[0];
+                ctx.bunVM().runErrorHandlerWithDedupe(err, null);
+                scope.done_result = err;
+            } else {
+                scope.done_result = .zero;
+            }
+        }
+
+        return JSValue.jsUndefined();
+    }
+
     pub fn execCallback(this: *DescribeScope, ctx: js.JSContextRef, comptime hook: LifecycleHook) JSValue {
         const name = comptime @as(string, @tagName(hook));
         var hooks: []JSC.JSValue = @field(this, name).items;
         for (hooks) |cb, i| {
             if (cb.isEmpty()) continue;
 
-            const err = cb.call(ctx, &.{});
-            if (err.isAnyError(ctx)) {
-                return err;
+            const pending_test = Jest.runner.?.pending_test;
+            // forbid `expect()` within hooks
+            Jest.runner.?.pending_test = null;
+            var result = if (cb.getLengthOfArray(ctx) > 0) brk: {
+                this.done_result = null;
+                const done_func = JSC.NewFunctionWithData(
+                    ctx,
+                    ZigString.static("done"),
+                    0,
+                    DescribeScope.onDone,
+                    false,
+                    this,
+                );
+                _ = cb.call(ctx, &.{done_func});
+                var vm = VirtualMachine.get();
+                while (true) {
+                    if (this.done_result) |result| {
+                        break :brk result;
```

Comment:
this would allow you to do

```js
beforeAll((done) => {
  done(fetch("https://example.com"))
});
```

and it would wait for fetch to fulfill

which is kind of interesting

but I don't think jest does that

Reply:
Indeed not &minus; that value is solely for error reporting, now fixed.